lib/rules/forbid-foreign-prop-types.js

```diff
@@ -34,26 +48,58 @@ module.exports = {
       return node.parent.type === 'AssignmentExpression' && node.parent.left === node;
     }
 
+    function isInPropTypesAssignment(node) {
+      if (!allowInPropTypes) {
+        return false;
+      }
+
+      if (node.parent.parent.parent.parent.type === 'AssignmentExpression') {
+        const assignmentExpression = node.parent.parent.parent.parent;
```

Comment:
might as well create this var above the conditional, so it can be reused in there?